include/language.php

```diff
@@ -148,78 +180,234 @@ class PLL_Language {
+	 *     tl_term_taxonomy_id?: positive-int,
+	 *     tl_count?: int<0, max>,
+	 *     term_props?: array{
+	 *         language?: array{
+	 *             term_id?: positive-int,
+	 *             term_taxonomy_id?: positive-int,
+	 *             count?: int<0, max>
+	 *         },
+	 *         term_language?: array{
+	 *             term_id?: positive-int,
+	 *             term_taxonomy_id?: positive-int,
+	 *             count?: int<0, max>
+	 *         }
+	 *     },
+	 *     name: non-empty-string,
+	 *     slug: non-empty-string,
+	 *     locale: non-empty-string,
+	 *     w3c: non-empty-string,
+	 *     flag_code: non-empty-string,
+	 *     term_group: int,
+	 *     is_rtl: int<0, 1>,
+	 *     mo_id?: positive-int,
+	 *     facebook?: non-empty-string,
+	 *     home_url?: non-empty-string,
+	 *     search_url?: non-empty-string,
+	 *     host?: non-empty-string,
+	 *     flag_url?: non-empty-string,
+	 *     flag?: non-empty-string,
+	 *     custom_flag_url?: non-empty-string,
+	 *     custom_flag?: non-empty-string,
+	 *     page_on_front?:positive-int,
+	 *     page_for_posts?:positive-int
+	 * } $language_data
+	 */
+	public function __construct( array $language_data ) {
+		$term_props = array(
+			'term_id'             => array( 'language', 'term_id' ),
+			'term_taxonomy_id'    => array( 'language', 'term_taxonomy_id' ),
+			'count'               => array( 'language', 'count' ),
+			'tl_term_id'          => array( 'term_language', 'term_id' ),
+			'tl_term_taxonomy_id' => array( 'term_language', 'term_taxonomy_id' ),
+			'tl_count'            => array( 'term_language', 'count' ),
+		);
+
+		// Deal with duplicated data: priority to the ones in `$this->term_props`.
+		foreach ( $term_props as $prop => $value ) {
+			if ( isset( $language_data['term_props'][ $value[0] ][ $value[1] ] ) ) {
+				$this->$prop = $this->set_tax_prop( $value[0], $value[1], $language_data['term_props'][ $value[0] ][ $value[1] ] );
+			} elseif ( isset( $language_data[ $prop ] ) ) {
+				$this->$prop = $this->set_tax_prop( $value[0], $value[1], $language_data[ $prop ] );
+			}
+
+			unset( $language_data['term_props'][ $value[0] ][ $value[1] ], $language_data[ $prop ] );
+		}
+
+		// Other values in `term_props` that are not `language` nor `term_language`.
+		if ( ! empty( $language_data['term_props'] ) ) {
+			foreach ( $language_data['term_props'] as $taxonomy_name => $prop_values ) {
+				foreach ( $prop_values as $prop_name => $prop_value ) {
+					$this->set_tax_prop( $taxonomy_name, $prop_name, $prop_value );
+				}
+			}
+		}
+
+		unset( $language_data['term_props'] );
+
+		// Add all the other values.
+		foreach ( $language_data as $prop => $value ) {
+			$this->$prop = $value;
+		}
+	}
+
+	/**
+	 * Returns a language term property value (like term ID, term taxonomy ID, or count).
+	 *
+	 * @since 3.4
+	 *
+	 * @param string $taxonomy_name Name of the taxonomy.
+	 * @param string $prop_name     Name of the property: 'term_taxonomy_id', 'term_id', 'count'.
```

Comment:
```suggestion
	 * @param string $prop_name     Name of the property (e.g. 'term_taxonomy_id', 'term_id', 'count').
```
Right now, I understand it accepts only 'term_taxonomy_id', 'term_id', 'count'.

Reply:
Indeed, it only accepts those.

Reply:
🙈